packages/manager/src/views/WebPropertyDetailPage/components/SSR/ConfigureSSRForm.tsx

```diff
@@ -0,0 +1,273 @@
+import { useAddSsrSpaProperty } from '@app/services/ssr';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { Button, Form, FormGroup, Split, SplitItem, TextInput } from '@patternfly/react-core';
+import { AddCircleOIcon } from '@patternfly/react-icons';
+import { useEffect, useState } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { toast } from 'react-hot-toast';
+import * as yup from 'yup';
+
+type FormData = {
+  propertyIdentifier: string;
+  name: string;
+  path: string;
+  ref: string;
+  env: string;
+  identifier: string;
+  nextRef: string;
+  accessUrl: string;
+  updatedAt: string;
+  _id: number;
+  isSSR: boolean;
+  healthCheckPath: string;
+  config: {
+    [key: string]: string;
+  };
+  imageUrl: string;
+};
+
+const schema = yup.object().shape({
+  name: yup.string().required(),
+  path: yup.string().required(),
+  imageUrl: yup.string().required(),
+  healthCheckPath: yup.string().required()
+});
+
+type Props = {
+  onClose: () => void;
+  propertyIdentifier: string;
+  dataprops: FormData;
+};
+
+export const ConfigureSSRForm = ({
+  onClose,
+  propertyIdentifier,
+  dataprops
+}: Props): JSX.Element => {
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+    reset
+  } = useForm<FormData>({
+    resolver: yupResolver(schema)
+  });
+
+  const createSsrSpaProperty = useAddSsrSpaProperty(propertyIdentifier);
+
+  const [keyValuePairs, setKeyValuePairs] = useState([{ key: '', value: '' }]);
+  const handleAddKeyValuePair = () => {
+    setKeyValuePairs([...keyValuePairs, { key: '', value: '' }]);
+  };
+
+  const handleKeyValuePairChange = (index: number, key: string, value: string) => {
+    const newKeyValuePairs = [...keyValuePairs];
+    newKeyValuePairs[index] = { key, value };
+    setKeyValuePairs(newKeyValuePairs);
+  };
+
+  const onSubmit = async (data: FormData) => {
+    const json: Record<string, string> = {};
+    keyValuePairs.forEach(({ key, value }) => {
+      if (key && value) {
+        json[key] = value;
+      }
+    });
+    const temp = data;
```

Comment:
Why?

Reply:
Coz this is a object. So by nature of js this is just a reference copy. So temp changes data key also changes. So why?

Reply:
used spread operator, earlier did for linting